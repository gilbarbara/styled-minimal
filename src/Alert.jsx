import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';

import { px, themeGet } from './utils/helpers';
import { baseStyles, sizesAllPropTypes, variantPropTypes } from './utils/system';

import Box from './Box';

const styles = (props: Object): string => {
  const { size } = props;
  const { borderRadius, maxWidth, padding } = themeGet(props, 'alert');

  return css`
    ${baseStyles.variant};
    border-radius: ${px(borderRadius)};
    font-size: ${baseStyles.fontSize};
    line-height: ${baseStyles.lineHeight};
    max-width: ${px(maxWidth)};
    padding: ${px(padding[size][0])} ${px(padding[size][1])};
    width: 100%;

    a {
      ${baseStyles.color};
    }
  `;
};

const Alert = styled(Box).attrs({
  role: 'alert',
})`
  ${styles};
`;

Alert.displayName = 'Alert';

Alert.propTypes = {
  as: PropTypes.string,
  children: PropTypes.node.isRequired,
  dark: PropTypes.bool,
  outline: PropTypes.bool,
  /** button size */
  size: sizesAllPropTypes,
  /** button variant */
  variant: variantPropTypes,
  ...Box.propTypes,
};

Alert.defaultProps = {
  outline: false,
  size: 'md',
  variant: 'primary',
};

export default Alert;
