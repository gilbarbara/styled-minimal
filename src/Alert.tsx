import * as React from 'react';
import styled, { css } from 'styled-components';

import { getTheme, px } from './utils/helpers';
import {
  baseStyles,
  colorStyles,
  lineHeightStyles,
  sx,
  system,
  variantStyles,
} from './utils/system';

import { AlertProps } from './types';

export const StyledAlert = styled('div')<AlertProps>(
  props => {
    const { borderRadius, maxWidth, padding } = getTheme(props, 'alert');

    return css`
      ${baseStyles};
      ${variantStyles};
      border-radius: ${px(borderRadius)};
      font-size: inherit;
      line-height: ${lineHeightStyles};
      max-width: ${px(maxWidth)};
      padding: ${padding};
      width: 100%;

      a {
        ${colorStyles};
      }
    `;
  },
  sx,
  system,
);

export const Alert = React.forwardRef<HTMLDivElement, AlertProps>((props, ref) => (
  <StyledAlert ref={ref} {...props} />
));

Alert.defaultProps = {
  dark: false,
  invert: false,
  role: 'alert',
  variant: 'primary',
};
