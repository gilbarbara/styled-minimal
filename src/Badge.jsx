import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';

import { px, themeGet } from './utils/helpers';
import { baseStyles, sizesAllPropTypes, variantPropTypes } from './utils/system';

import Box from './Box';

export const Badge = styled(Box)`
  ${(props: Object): string => {
    const {
      borderRadius,
      fontSize,
      fontWeight,
      padding,
    } = themeGet(props, 'badge');
    const fontSizeProp = themeGet(props, 'componentSizes', { key: 'size' });

    return css`
      ${baseStyles.variant};
      border-radius: ${px(borderRadius)};
      display: inline-flex;
      font-size: ${fontSize};
      ${fontSizeProp ? `font-size: ${fontSizeProp}` : ''};
      font-weight: ${fontWeight};
      line-height: 1;
      padding: ${px(padding[0])} ${px(padding[1])};
      vertical-align: baseline;
    `;
  }};
`;

Badge.displayName = 'Badge';

Badge.propTypes = {
  as: PropTypes.string,
  children: PropTypes.node.isRequired,
  dark: PropTypes.bool,
  outline: PropTypes.bool,
  size: sizesAllPropTypes,
  variant: variantPropTypes,
  ...Box.propTypes,
};

Badge.defaultProps = {
  as: 'span',
};

export default Badge;
