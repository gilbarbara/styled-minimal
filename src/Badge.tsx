import * as React from 'react';
import styled, { css } from 'styled-components';

import { getTheme, px } from './utils/helpers';
import { baseStyles, sx, system, variantStyles } from './utils/system';

import { BadgeProps } from './types';

export const StyledBadge = styled('sup')<BadgeProps>(
  props => {
    const { dot } = props;
    const { borderRadius, fontSize, fontWeight, lineHeight, padding, size, sizeDot } = getTheme(
      props,
      'badge',
    );

    return css`
      ${baseStyles};
      ${variantStyles};
      border-radius: ${dot ? '100%' : px(borderRadius)};
      display: inline-flex;
      font-size: ${dot ? 0 : px(fontSize)};
      font-weight: ${fontWeight};
      justify-content: center;
      line-height: ${dot ? 0 : px(lineHeight)};
      height: ${px(dot ? sizeDot : size)};
      padding: ${dot ? 0 : `0 ${px(padding)}`};
      min-width: ${px(dot ? sizeDot : size)};
    `;
  },
  sx,
  system,
);

export const Badge = React.forwardRef<HTMLSpanElement, BadgeProps>((props, ref) => (
  <StyledBadge ref={ref} {...props} />
));

Badge.defaultProps = {
  dark: false,
  dot: false,
  invert: false,
  variant: 'primary',
};
