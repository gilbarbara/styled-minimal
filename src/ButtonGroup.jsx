import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import { sizesAllPropTypes, variantPropTypes } from './utils/system';

import Box from './Box';

export const StyledButtonGroup = styled(Box)`
  display: inline-flex;

  > button {
    + button {
      margin-left: -1px;
    }

    &:first-child {
      border-bottom-right-radius: 0;
      border-top-right-radius: 0;
    }

    &:last-child {
      border-bottom-left-radius: 0;
      border-top-left-radius: 0;
    }

    &:not(:first-child):not(:last-child) {
      border-radius: 0;
    }
  }
`;

const ButtonGroup = ({ children, size, variant, ...props }) => {
  const buttonProps = {};

  if (typeof size !== 'undefined') {
    buttonProps.size = size;
  }

  if (typeof variant !== 'undefined') {
    buttonProps.variant = variant;
  }

  return (
    <StyledButtonGroup {...props}>
      {React.Children.map(children, child => React.cloneElement(child, { ...buttonProps }))}
    </StyledButtonGroup>
  );
};

ButtonGroup.displayName = 'ButtonGroup';

ButtonGroup.propTypes = {
  children: PropTypes.node.isRequired,
  size: sizesAllPropTypes,
  variant: variantPropTypes,
  ...Box.basePropTypes,
};

export default ButtonGroup;
