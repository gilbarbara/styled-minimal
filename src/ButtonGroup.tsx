import * as React from 'react';
import styled, { css } from 'styled-components';

import { baseStyles, sx, system } from './utils/system';

import { ButtonGroupProps } from './types';

export const StyledButtonGroup = styled('div')<Partial<ButtonGroupProps>>(
  () => css`
    ${baseStyles};
    display: inline-flex;

    > button {
      + button {
        margin-left: -1px;
      }

      &:first-child {
        border-bottom-right-radius: 0;
        border-top-right-radius: 0;
      }

      &:last-child {
        border-bottom-left-radius: 0;
        border-top-left-radius: 0;
      }

      &:not(:first-child):not(:last-child) {
        border-radius: 0;
      }
    }
  `,
  sx,
  system,
);

export const ButtonGroup = React.forwardRef<HTMLDivElement, ButtonGroupProps>(
  ({ children, size, variant, ...rest }: ButtonGroupProps, ref) => {
    const buttonProps = {
      size,
      variant,
    };

    return (
      <StyledButtonGroup ref={ref} {...rest}>
        {React.Children.map(children, child =>
          React.cloneElement(child as React.ReactElement, { ...buttonProps }),
        )}
      </StyledButtonGroup>
    );
  },
);

ButtonGroup.defaultProps = {
  size: 'md',
  variant: 'primary',
};
