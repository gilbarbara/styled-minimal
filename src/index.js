export { default as Alert } from './Alert';
export { default as Badge } from './Badge';
export { default as Box } from './Box';
export { default as Button } from './Button';
export { default as Code } from './Code';
export { default as Container } from './Container';
export { default as Embed } from './Embed';
export { default as Flex } from './Flex';
export { default as Fieldset } from './Fieldset';
export { default as Form } from './Form';
export { default as FormGroup } from './FormGroup';
export { default as Group } from './Group';
export { default as Heading } from './Heading';
export { default as Image } from './Image';
export { default as Input } from './Input';
export { default as Label } from './Label';
export { default as Legend } from './Legend';
export { default as Link } from './Link';
export { default as List } from './List';
export { default as Paragraph } from './Paragraph';
export { default as Screen } from './Screen';
export { default as Select } from './Select';
export { default as Switch } from './Switch';
export { default as Table } from './Table';
export { default as Text } from './Text';
export { default as Textarea } from './Textarea';
