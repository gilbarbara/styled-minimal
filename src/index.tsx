export { Alert } from './Alert';
export { Badge } from './Badge';
export { Box } from './Box';
export { Button } from './Button';
export { ButtonGroup } from './ButtonGroup';
export { Code } from './Code';
export { CodeBlock } from './CodeBlock';
export { Container } from './Container';
export { Embed } from './Embed';
export { Flex } from './Flex';
export { Fieldset } from './Fieldset';
export { Form } from './Form';
export { FormGroup } from './FormGroup';
export { Grid } from './Grid';
export { Group } from './Group';
export { Heading } from './Heading';
export { Image } from './Image';
export { Checkbox, Input, Radio } from './Input';
export { Label } from './Label';
export { Legend } from './Legend';
export { Link } from './Link';
export { DL, OL, UL } from './List';
export { Loader } from './Loader';
export { Paragraph } from './Paragraph';
export { RadioGroup } from './RadioGroup';
export { Select } from './Select';
export { Switch } from './Switch';
export { Table } from './Table';
export { Text } from './Text';
export { Textarea } from './Textarea';

export { default as system } from '@styled-system/css';
export { mergeTheme as getTheme, px, responsive } from './utils/helpers';
