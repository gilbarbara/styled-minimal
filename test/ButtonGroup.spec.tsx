import * as React from 'react';
import { render } from '@testing-library/react';

import { ButtonGroup } from '../src/ButtonGroup';
import { Button } from '../src';

describe('ButtonGroup', () => {
  const { container } = render(
    <ButtonGroup>
      <Button invert>First</Button>
      <Button>Second</Button>
      <Button invert>Third</Button>
      <Button invert>Forth</Button>
    </ButtonGroup>,
  );

  it('should render properly', () => {
    expect(container.firstChild).toMatchSnapshot();
  });
});
