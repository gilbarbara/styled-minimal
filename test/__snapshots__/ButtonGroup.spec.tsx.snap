// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonGroup should render properly 1`] = `
.c0 {
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.c0 > button + button {
  margin-left: -1px;
}

.c0 > button:first-child {
  border-bottom-right-radius: 0;
  border-top-right-radius: 0;
}

.c0 > button:last-child {
  border-bottom-left-radius: 0;
  border-top-left-radius: 0;
}

.c0 > button:not(:first-child):not(:last-child) {
  border-radius: 0;
}

.c1 {
  box-sizing: border-box;
  background-color: transparent;
  border: 1px solid #2E2E2E;
  color: #2E2E2E;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 0px;
  box-shadow: none;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-family: inherit;
  font-size: 16px;
  line-height: 1;
  padding: 8px 14px;
  -webkit-text-decoration: none;
  text-decoration: none;
  width: auto;
}

.c1:disabled {
  opacity: 0.7;
  pointer-events: none;
}

.c1:focus {
  outline-color: #2E2E2E;
}

.c1 .c3 {
  margin-left: 5px;
}

.c2 {
  box-sizing: border-box;
  background-color: #2E2E2E;
  border: 1px solid #2E2E2E;
  color: #fff;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 0px;
  box-shadow: none;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-family: inherit;
  font-size: 16px;
  line-height: 1;
  padding: 8px 14px;
  -webkit-text-decoration: none;
  text-decoration: none;
  width: auto;
}

.c2:disabled {
  opacity: 0.7;
  pointer-events: none;
}

.c2:focus {
  outline-color: #2E2E2E;
}

.c2 .c3 {
  margin-left: 5px;
}

<div
  class="c0"
>
  <button
    class="c1"
    type="button"
  >
    First
  </button>
  <button
    class="c2"
    type="button"
  >
    Second
  </button>
  <button
    class="c1"
    type="button"
  >
    Third
  </button>
  <button
    class="c1"
    type="button"
  >
    Forth
  </button>
</div>
`;
